### ISSUE TITLE
when use env variable  due to make compiler  exit!

### ISSUE THREAD

**Issue by xicaixiaokeke** [dagu-dev/dagu#475]
GCC version:gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)
dagu version: 1.11.0

**_[Use env variable make build occur ERROR Message]_**
make -C /lib/modules/3.10.0-1160.76.1.el7.x86_64/build M=/tmp/bbc-patch/071_build_bnxt_driver/centos/bnxt_build modules
make[1]: Entering directory `/usr/src/kernels/3.10.0-1160.76.1.el7.x86_64'
arch/x86/Makefile:96: stack-protector enabled but compiler support broken
arch/x86/Makefile:166: *** CONFIG_RETPOLINE=y, but not supported by the compiler. Compiler update recommended..  Stop.
make[1]: Leaving directory `/usr/src/kernels/3.10.0-1160.76.1.el7.x86_64'
make: *** [default] Error 2
+ '[' 2 -ne 0 ']'
+ echo 'Build bnxt_en with kernel 3.10.0-1160.76.1.el7.x86_64 failed. Please check ...'
Build bnxt_en with kernel 3.10.0-1160.76.1.el7.x86_64 failed. Please check ...
+ exit 1

**_[Does not use env variable make compiler is normal ]_**
make -C /lib/modules/3.10.0-1160.76.1.el7.x86_64/build M=/tmp/bbc-patch/071_build_bnxt_driver/centos/bnxt_build modules
make[1]: Entering directory `/usr/src/kernels/3.10.0-1160.76.1.el7.x86_64'
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /tmp/bbc-patch/071_build_bnxt_driver/centos/bnxt_build/bnxt_en.mod.o
  LD [M]  /tmp/bbc-patch/071_build_bnxt_driver/centos/bnxt_build/bnxt_en.ko
make[1]: Leaving directory `/usr/src/kernels/3.10.0-1160.76.1.el7.x86_64'
+ '[' 0 -ne 0 ']'
+ make KVER=3.10.0-1160.76.1.el7.x86_64 install
make -C /lib/modules/3.10.0-1160.76.1.el7.x86_64/build M=/tmp/bbc-patch/071_build_bnxt_driver/centos/bnxt_build modules
make[1]: Entering directory `/usr/src/kernels/3.10.0-1160.76.1.el7.x86_64'
  Building modules, stage 2.
  MODPOST 1 modules

**Comment by xicaixiaokeke** [comment]
dag spec 
env:
   - DIR: /tmp

**Comment by bbqi** [comment]
use env NOT Add arguments to environment variables.
it'll create NEW environment. NOT contain origin system variables.

dag spec:
```
#env:
#  - DIR: /tmp
steps:
  - name: step1
    command: env
```
then try remove comments, results are different.

**Comment by yohamta** [comment]
Thanks a lot for reporting the issue. This issue is fixed in v1.13.0.
